src/main/java/io/aiven/kafka/connect/opensearch/RecordConverter.java

```diff
@@ -133,6 +149,29 @@ private String getPayload(final SinkRecord record) {
         return new String(rawJsonPayload, StandardCharsets.UTF_8);
     }
 
+    private String addTimestampToPayload(final String payload, final long timestamp) {
+        if (DATA_STREAM_TIMESTAMP_FIELD_DEFAULT.equals(config.dataStreamTimestampField())) {
```

Comment:
I would suggested to drop this condition and always make sure `config.dataStreamTimestampField()` is present in the payload

Reply:
in case of missing a custom timetamp field OpenSearch throws exception